src/plugins/query_enhancements/common/utils.ts

```diff
@@ -57,13 +58,34 @@ export const fetch = (context: EnhancedFetchContext, query: Query, aggConfig?: Q
     pollQueryResultsParams: context.body?.pollQueryResultsParams,
     timeRange: context.body?.timeRange,
   });
+
   return from(
-    http.fetch({
-      method: 'POST',
-      path,
-      body,
-      signal,
-    })
+    http
+      .fetch({
+        method: 'POST',
+        path,
+        body,
+        signal,
+      })
+      .catch(async (error) => {
+        if (error.name === 'AbortError' && context.body?.pollQueryResultsParams?.queryId) {
```

Comment:
i can't remember but does the current UI abort requests when user clicks Run query before the previous one finishes?

For example, 

1. user starts async query
2. query id returned
3. browser sends requests to check job status
4. server returns job status
5. browser waits a few seconds if job is pending
6. step 3-5 repeats until query finishes

If the request in step 3 is aborted, then this would work. But if user runs another query during step 5, would this abort and delete job for the previous query?

Reply:
In this scenario, yes, it will abort if another query is run in between the short polling requests.

Reply:
could you explain a bit why would it abort? i thought during waiting time, the last `fetch` promise will be resolved and `catch` won't trigger?